File: app/assets/javascripts/app/models/meme.js
App.Models.Meme = Backbone.Model.extend({
  defaults: {
    top_text: '',
    bottom_text: '',
    image: '',
    meme_template_name: 'xzibit'
  },
  url: function(){
    if(this.isNew()){
      return '/memes';
    }else{
      return '/memes/' + this.id;
    }
  }
})

File: app/models/meme.rb
class Meme < ActiveRecord::Base
  has_attached_file :image, :styles => { :medium => "300x300>", :thumb => "100x100>" }  

  def attributes_for_json
    self.slice(:top_text, :bottom_text, :id, :meme_template_name).merge(:image => image.url, :image_thumb => image.url(:thumb))
  end
end
